### PR TITLE
STRK-196: align trade cost-basis divisor with actually-linked count (no-bump — release artifacts deferred to #1298)

### PR DESCRIPTION
**Pull request by lbruton**
## STRK-196 — trade cost-basis divisor

**Problem.** `linkTradeItems()` (`js/inventory.js`) divided the given-up cost basis by the **raw** `receivedUuids.length`, but the apply loop skips falsy, duplicate, self, unresolvable, and reassignment-declined entries. Any such entry inflated the divisor → each received item got a **diluted (too-low) cost basis**, and a duplicate uuid emitted a **redundant trade-link change-log row**.

**Fix.** Extract `_resolveTradeCandidates()` to resolve the final accepted set (dedupe + filter self/missing + resolve reassignment prompts) **before** any link is applied, then use `accepted.length` as the divisor. Clean unique input is byte-for-byte unchanged; `_applyTradeLink()` (and its STRK-132 carryover semantics) is untouched.

**Test.** New core disposition regression direct-calls `window.linkTradeItems()` with `["received-one","received-one","",source,"missing","received-two"]` and asserts: only the two unique items linked, each price `* 2 ≈ givenUpValue` (split by 2, not the raw 6), and exactly 2 trade-link change-log rows. Red on the old divisor, green on the fix.

## ⚠️ Sequencing — this PR is now NO-BUMP

This was one of three concurrent version-lock PRs off the same `3.35.31` base (#1296 v3.35.32 STRK-203 → merged; this #1297; #1298 v3.35.34 STRK-204). To avoid resolving the CHANGELOG / What's New stacking conflict on every PR, the **version bump + release notes were stripped from this PR** and merged clean against `dev` @ 3.35.32. They will be **consolidated into #1298** (the last to land) so the release-file stacking happens once. Tracked as **DEVS-59**.

**This PR's diff vs `dev`:** `js/inventory.js` (fix), `tests/playwright/core/disposition.spec.js` (regression), `tests/playwright/coverage-map.csv` (trade-linking 4→5), and a benign `sw.js` cache-epoch restamp. **No version bump.**

### For #1298 to absorb (STRK-196 release notes)

CHANGELOG section:
```
### Changed — STRK-196: Accurate trade cost basis for multi-item trades

- **Disposition → Trade links**: When trading one item for multiple received items, the given-up cost basis is now divided by the number of items actually linked, not the raw input length. Duplicate, empty, self-referential, or unresolvable entries no longer dilute each received item's cost basis or emit redundant trade-link change-log rows. Everyday trades are unchanged; this corrects an edge case in cost-basis allocation (STRK-196).
```

What's New entry (use #1298's assigned version):
```
<li><strong>vX.XX.XX &ndash; STRK-196: Accurate trade cost basis for multi-item trades</strong>: When you trade one item for several received items, the carried-over cost basis is now split across only the items actually linked. Previously, duplicate or invalid entries in a trade could inflate the divisor and leave each received item with a too-low cost basis &mdash; everyday trades were unaffected, but this hardens the edge cases (STRK-196).</li>
```

## Verification
- `npm run test:core` → 289 passed pre-merge; 5 trade-linking green on the merged tree.
- `npm run lint` → clean.
- coverage-map trade-linking count 4 → 5.

Closes STRK-196.

### CODE REVIEW

File: js/inventory.js [hunk 1]
```diff
@@ -986,7 +996,25 @@ const linkTradeItems = async (disposedItem, receivedUuids, tradeDate) => {
...
+  if (!disposedItem?.disposition || !Array.isArray(receivedUuids)) return [];
+  if (!Array.isArray(disposedItem.disposition.tradedForUuids)) {
+    disposedItem.disposition.tradedForUuids = [];
+  }
+  if (!disposedItem.disposition.tradeValues) disposedItem.disposition.tradeValues = {};
+  // STRK-196: resolve the final accepted set first so the cost-basis divisor
+  // equals the number of items actually linked. Duplicate/falsy/self/missing/
+  // declined entries must not dilute per-item price or duplicate change-log rows.
+  const accepted = await _resolveTradeCandidates(disposedItem, receivedUuids);
+  const receivedCount = accepted.length;
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/inventory.js:1014]: Valid observation, but this is a **pre-existing condition and out of scope for STRK-196**. STRK-196 targets the *within-call* divisor/loop mismatch (duplicate/falsy/self/missing entries in a single `linkTradeItems` call). The batch-vs-total issue you describe lives in the **edit-an-existing-trade** path: `updateTradeLinks` passes only the newly-added uuids to `linkTradeItems`, so the divisor is the batch size — and the old code did exactly the same (`receivedUuids.length` == `added.length`). A correct fix must re-price the *full* linked set so the allocation sums to the given-up value, which is a behavior change beyond this PR's scope. Tracked as **STRK-229**.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: js/inventory.js [hunk 2]
```diff
@@ -986,7 +996,25 @@ const linkTradeItems = async (disposedItem, receivedUuids, tradeDate) => {
         typeof findItemByUuid === "function" ? findItemByUuid(receivedItem.tradedFromUuid) : null;
       removeTradeLinkReference(oldSource, receivedUuid);
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/inventory.js]: Added an explicit `if (oldSource)` guard at the call site in 9a7e6524. The callee `removeTradeLinkReference` already early-returns on a null first arg (`if (!disposedItem?.disposition) return;`), so there was no actual runtime-exception risk — but the explicit guard makes the call-site intent self-documenting.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: sw.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ importScripts("sw-router.js");
 
 const DEV_MODE = false; // Set to true during development — bypasses all caching
 
-const CACHE_NAME = "staktrakr-v3.35.32-b1781826533";
+const CACHE_NAME = "staktrakr-v3.35.32-b1781828379";
```
[the remark this answers is not in the document]
Comment (lbruton) [reply sw.js]: Intentional — this is now a **no-bump PR**. The version bump and all release artifacts (`constants.js`, `version.json`, `package.json`, the `about.js` What's New entry, and the `sw.js` stamp) were deliberately stripped and deferred to **PR #1298**, which consolidates the release notes for three concurrent version-lock PRs so the CHANGELOG / What's New stacking is resolved once instead of conflicting on every PR (see **DEVS-59**). The `patch/3.35.33` branch name is historical; STRK-196's release metadata lands via #1298.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
